node/narwhal/src/helpers/committee.rs

```diff
@@ -324,118 +324,174 @@ pub mod prop_tests {
     pub struct Validator {
-        #[strategy(..5_000_000_000u64)]
         pub stake: u64,
-        account_seed: u64,
+        pub account: Account<CurrentNetwork>,
     }
 
-    // Validators can have the same stake but shouldn't have the same account seed.
-    impl PartialEq for Validator {
-        fn eq(&self, other: &Self) -> bool {
-            self.account_seed == other.account_seed
+    impl Arbitrary for Validator {
+        type Parameters = ();
+        type Strategy = BoxedStrategy<Validator>;
+
+        fn arbitrary_with(_: Self::Parameters) -> Self::Strategy {
+            any_valid_validator()
         }
     }
 
-    // Make sure the Hash matches PartialEq.
-    impl Hash for Validator {
-        fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
-            self.account_seed.hash(state);
+    fn to_committee((round, ValidatorSet(validators)): (u64, ValidatorSet)) -> Result<Committee<CurrentNetwork>> {
+        Committee::new(round, validators.iter().map(|v| (v.account.address(), v.stake)).collect())
+    }
+
+    fn validator_set<T: Strategy<Value = Validator>>(
+        element: T,
+        size: impl Into<SizeRange>,
+    ) -> impl Strategy<Value = ValidatorSet> {
+        hash_set(element, size).prop_map(ValidatorSet)
+    }
+
+    fn invalid_round_committee() -> BoxedStrategy<Result<Committee<CurrentNetwork>>> {
+        (Just(0), validator_set(any_valid_validator(), size_range(4..=MAX_COMMITTEE_SIZE as usize)))
+            .prop_map(to_committee)
+            .boxed()
+    }
+
+    fn too_small_committee() -> BoxedStrategy<Result<Committee<CurrentNetwork>>> {
+        (1u64.., validator_set(any_valid_validator(), 0..4)).prop_map(to_committee).boxed()
+    }
+
+    fn too_low_stake_committee() -> BoxedStrategy<Result<Committee<CurrentNetwork>>> {
+        (1u64.., validator_set(invalid_stake_validator(), 4..=4)).prop_map(to_committee).boxed()
+    }
+
+    #[derive(Debug, Clone)]
+    pub struct CommitteeContext(pub Committee<CurrentNetwork>, pub ValidatorSet);
+
+    impl Default for CommitteeContext {
+        fn default() -> Self {
+            let validators = ValidatorSet::default();
+            let committee = to_committee((u64::default(), validators.clone())).unwrap();
+            Self(committee, validators)
         }
     }
 
-    impl Validator {
-        pub fn get_account(&self) -> Account<CurrentNetwork> {
-            match Account::new(&mut rand_chacha::ChaChaRng::seed_from_u64(self.account_seed)) {
-                Ok(account) => account,
-                Err(err) => panic!("Failed to create account {err}"),
-            }
+    #[derive(Debug, Clone)]
+    pub struct ValidatorSet(pub HashSet<Validator>);
+
+    impl Default for ValidatorSet {
+        fn default() -> Self {
+            ValidatorSet(
+                (0..=4u64)
+                    .map(|i| Validator {
+                        account: Account::new(&mut rand_chacha::ChaChaRng::seed_from_u64(i)).unwrap(),
```

Comment:
```suggestion
                        account: Account::new(&mut TestRng::fixed(i)).unwrap(),
```

Reply:
https://github.com/AleoHQ/snarkOS/pull/2537#discussion_r1264311767 and https://github.com/AleoHQ/snarkOS/pull/2537#discussion_r1264310224